src/v0/destinations/am/transform.js

```diff
@@ -545,26 +610,30 @@ function processSingleMessage(message, destination) {
           .trim();
         evType =
           userProvidedPageEventString.trim() === ''
-            ? message.name
+            ? name
             : userProvidedPageEventString
                 .trim()
                 .replaceAll(/{{([^{}]+)}}/g, get(message, getMessagePath));
       } else {
-        evType = `Viewed ${message.name || get(message, CATEGORY_KEY) || ''} Page`;
+        const updatedName = getUpdatedPageNameWithoutUserDefinedPageEventName(
+          name,
+          message,
+          CATEGORY_KEY,
+        );
+        evType = `Viewed ${updatedName || ''}Page`;
       }
-
       message.properties = {
         ...message.properties,
-        name: message.name || get(message, CATEGORY_KEY),
+        name: name || get(message, CATEGORY_KEY),
       };
       category = ConfigCategory.PAGE;
       break;
     case EventType.SCREEN:
-      evType = `Viewed ${message.name || message.event || get(message, CATEGORY_KEY) || ''} Screen`;
-      message.properties = {
-        ...message.properties,
-        name: message.name || message.event || get(message, CATEGORY_KEY),
-      };
+      ({ evType, updatedProperties: properties } = getScreenevTypeAndUpdatedProperties(
+        message,
+        CATEGORY_KEY,
+      ));
+      message.properties = properties;
```

Comment:
@anantjain45823 can you check this ?
```suggestion
      message.properties = updatedProperties;
```

Reply:
@krishna2020 `updatedProperties` is the object returned from the function which we are mapping to `properties` at [this point ](https://github.com/rudderlabs/rudder-transformer/blob/11fb7c47910681833e37d25a1573d2005e62742b/src/v0/destinations/am/transform.js#L632).
Hence we are using `properties` and not `updatedProperties`

Reply:
when there's already variable `updatedProperties ` defined, why not use it directly instead of assigning it to `properties` again ? It's simplifies the readability.

({ evType, updatedProperties } = getScreenevTypeAndUpdatedProperties(
        message,
        CATEGORY_KEY,
      ));
      message.properties = updatedProperties;

Reply:
Done